builtin/credential/approle/backend.go

```diff
@@ -73,6 +74,10 @@ func Backend(conf *logical.BackendConfig) (*backend, error) {
 	// a superfluous number of locks directly proportional to the number of RoleID
 	// and SecretIDs. These locks can be accessed by indexing based on the first two
 	// characters of a randomly generated UUID.
+	if err = locksutil.CreateLocks(b.roleLocksMap, 256); err != nil {
```

Comment:
Might actually be nice to have `locksutil` check if the map is nil and do a `make` in there with the desired size if so.